apps/prairielearn/src/pages/instructorAssessmentInstances/instructorAssessmentInstances.ts

```diff
@@ -111,20 +112,23 @@ router.post(
         assessment_instance_id: req.body.assessment_instance_id,
         assessment_id: res.locals.assessment.id,
         time_add: req.body.time_add,
-        time_ref: req.body.time_ref,
         base_time: 'date_limit',
         authn_user_id: res.locals.authz_data.authn_user.user_id,
+        exact_date: formatDate(new Date(), res.locals.course_instance.display_timezone),
       };
-      if (req.body.plus_minus === 'unlimited') {
+      if (req.body.plus_minus === 'unlimited' || req.body.reopen_without_limit === 'true') {
         params.base_time = 'null';
       } else if (req.body.plus_minus === 'expire') {
         params.base_time = 'current_date';
         params.time_add = 0;
-        params.time_ref = 'minutes';
       } else if (req.body.plus_minus === 'set_total') {
         params.base_time = 'start_date';
       } else if (req.body.plus_minus === 'set_rem') {
         params.base_time = 'current_date';
+      } else if (req.body.plus_minus === 'set_exact') {
+        params.base_time = 'exact_date';
+        params.time_add = 0;
+        params.exact_date = `${req.body.date} ${res.locals.course_instance.display_timezone}`;
```

Comment:
Looks like this is sort of implicitly relying on the way that Postgres parses dates. Can we do that parsing in JS somehow?

Reply:
Updated to use Temporal to parse the date in [42543c9](https://github.com/PrairieLearn/PrairieLearn/pull/10800/commits/42543c9a1ca769d8b9f72a66c1f6d1916060f477).

Reply:
Looks good, but you missed one farther down in the file.

Reply:
Updated that one in [edf82d6](https://github.com/PrairieLearn/PrairieLearn/pull/10800/commits/edf82d688b0926f851ceb946a36c6cfce3f4b686).